File: api/modules/notification/notificationModel/notification_model.go
package notificationModel

import (
	"time"

	"github.com/khiemnd777/noah_api/shared/db/ent/generated"
)

type Notification struct {
	ID         int             `json:"id,omitempty"`
	UserID     int             `json:"user_id,omitempty"`
	NotifierID int             `json:"notifier_id,omitempty"`
	CreatedAt  time.Time       `json:"created_at,omitempty"`
	Type       string          `json:"type,omitempty"`
	Read       bool            `json:"read,omitempty"`
	Data       map[string]any  `json:"data,omitempty"`
	Notifier   *generated.User `json:"notifier,omitempty"`
}

File: fe/src/core/notification/notification.model.ts
import type { UserModel } from "../auth/auth.types";

export interface NotificationModel<T = any> {
  id?: number;
  userId?: number;
  notifierId?: number;
  createdAt?: string;
  type?: string;
  read?: boolean;
  readAt?: string | null;
  body?: string;
  data?: T;
  notifier?: UserModel | null;
}
